internal/controller/warehouses/git.go

```diff
@@ -303,75 +306,97 @@ func ignores(tagName string, ignore []string) bool {
 func matchesPathsFilters(includePaths []string, excludePaths []string, diffs []string) (bool, error) {
-	includePathsRegexps, err := compileRegexps(includePaths)
+	includePathsRegexps, includePathsGlobs, includePathsPrefixes, err := sortFilters(includePaths)
 	if err != nil {
 		return false, fmt.Errorf(
 			"error compiling includePaths regexps: %w",
 			err,
 		)
 	}
 
-	excludePathsRegexps, err := compileRegexps(excludePaths)
+	excludePathsRegexps, excludePathsGlobs, excludePathsPrefixes, err := sortFilters(excludePaths)
 	if err != nil {
 		return false, fmt.Errorf(
 			"error compiling excludePaths regexps: %w",
 			err,
 		)
 	}
 
-	filteredDiffs := make([]string, 0, len(diffs))
 	for _, diffPath := range diffs {
+		matchesIncludeGlobs, err := matchesGlobList(diffPath, includePathsGlobs)
+		if err != nil {
+			return false, fmt.Errorf(
+				"syntax error in include glob patterns: %w",
+				err,
+			)
+		}
+		matchesExcludeGlobs, err := matchesGlobList(diffPath, excludePathsGlobs)
+		if err != nil {
+			return false, fmt.Errorf(
+				"syntax error in exclude glob patterns: %w",
+				err,
+			)
+		}
+		matchesIncludePrefixes := matchesPrefixList(diffPath, includePathsPrefixes)
+		matchesExcludePrefixes := matchesPrefixList(diffPath, excludePathsPrefixes)
 		// matchesIncludePaths case is a bit different from matchesExcludePaths
 		// in the way that if includePaths string array is empty - it matches
 		// ANY change so we need to have a check for that
-		matchesIncludePaths := len(includePaths) == 0 || matchesRegexpList(diffPath, includePathsRegexps)
-		matchesExcludePaths := matchesRegexpList(diffPath, excludePathsRegexps)
-		// combined filter decision, positive for matching includePaths and
-		// unmatching excludePaths
-		if matchesIncludePaths && !matchesExcludePaths {
-			filteredDiffs = append(filteredDiffs, diffPath)
+		matchesIncludePaths := len(includePaths) == 0 || matchesRegexpList(
+			diffPath,
+			includePathsRegexps,
+		) || matchesIncludeGlobs || matchesIncludePrefixes
+		if !matchesIncludePaths {
+			continue
+		}
+		matchesExcludePaths := matchesRegexpList(diffPath,
+			excludePathsRegexps,
+		) || matchesExcludeGlobs || matchesExcludePrefixes
+		if matchesExcludePaths {
+			continue
 		}
-	}
-	if len(filteredDiffs) > 0 {
 		return true, nil
 	}
 	return false, nil
 }
 
-// compileRegexps is a general purpose function taking a slice of strings
-// as argument and compiling them into a slice of *regexp.Regexp. It
-// returns the compiled regexps in case of success and if it fails to compile
-// a string - nil and error is returned
-func compileRegexps(regexpStrings []string) (regexps []*regexp.Regexp, err error) {
+// sortFilters handles sorting of a slice of strings to regexps,
+// globs and prefixes, regexps are compiled into a slice of
+// *regexp.Regexp additionally
+func sortFilters(regexpStrings []string) (regexps []*regexp.Regexp, globs []string, prefixes []string, err error) {
```

Comment:
`regexpStrings` is probably not an ideal name for this argument given that the input can also contain glob patterns and prefixes.

Note that we also don't use named returns anywhere else in the code base.